src/callback.rs

```diff
@@ -176,3 +177,169 @@ where
 {
     value.convert(py)
 }
+
+// Autoref-based specialization for handling `__next__` returning `Option`
```

Comment:
For what it's worth I was sort of hoping that in `src/callback.rs` we'll either have only public API or nothing at all. 

Would you be amenable to moving these types to `src/impl_/pymethods.rs`, maybe?

Reply:
Yes, no problem. Will do.